sources/main.c

```diff
@@ -22,8 +22,10 @@ int	main(void)
 		if (ft_strcmp(input, "exit") == 0)
 		{
 			write(1, "exit\n", 5);
+			clear_history();
 			break ;
 		}
+		add_history(input);
```

Comment:
Empty lines should not be saved to history. You could put a codition here to treat it.